From 4011177b8ea83c54da7823ea21d361a84967060f Mon Sep 17 00:00:00 2001
From: Huacai Chen <chenhc@lemote.com>
Date: Thu, 1 Dec 2016 09:51:35 +0800
Subject: [PATCH 087/136] Input: i8042 - Make i8042_bypass_aux_irq_test as a
 module parameter

The original i8042_bypass_aux_irq_test is a variable which cannot be
changed without kernel rebuild. Some buggy platform fails to detect the
existance of AUX if i8042_bypass_aux_irq_test isn't set. So make this
variable as a module parameter and set the default value to true.

Signed-off-by: Huacai Chen <chenhc@lemote.com>
---
 drivers/input/serio/i8042.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/input/serio/i8042.c b/drivers/input/serio/i8042.c
index 0de3d0b81..05fbd3379 100644
--- a/drivers/input/serio/i8042.c
+++ b/drivers/input/serio/i8042.c
@@ -126,7 +126,9 @@ MODULE_PARM_DESC(unmask_kbd_data, "Unconditional enable (may reveal sensitive da
 #endif
 
 static bool i8042_present;
-static bool i8042_bypass_aux_irq_test;
+static bool i8042_bypass_aux_irq_test = true;
+module_param_named(bypass_aux_test, i8042_bypass_aux_irq_test, bool, 0600);
+MODULE_PARM_DESC(debug, "Turn i8042 aux irq test off");
 static char i8042_kbd_firmware_id[128];
 static char i8042_aux_firmware_id[128];
 
-- 
2.39.1

